app/services/hyrax/admin_set_create_service.rb

```diff
@@ -91,28 +94,37 @@ def create_workflows_for(permission_template:)
         Sipity::Workflow.activate!(permission_template: permission_template, workflow_name: Hyrax.config.default_active_workflow_name)
       end
 
+      def register_managing_role!
+        # Force creation of registered MANAGING role if it doesn't exist
+        Sipity::Role[Hyrax::RoleRegistry::MANAGING]
+      end
+
       def grant_all_workflow_roles_to_creating_user_and_admins!(permission_template:)
-        # Default admin set has a nil creating_user; guard against that condition
-        return if creating_user.nil?
+        # This code must be invoked before calling `Sipity::Role.all` or the managing role won't be there
+        register_managing_role!
         # Grant all workflow roles to the creating_user and the admin group
         permission_template.available_workflows.each do |workflow|
           Sipity::Role.all.each do |role|
             workflow.update_responsibilities(role: role,
-                                             agents: [
-                                               creating_user,
-                                               Hyrax::Group.new(admin_group_name)
-                                             ])
+                                             agents: workflow_agents)
           end
         end
       end
 
-      # Gives deposit access to registered users and manage access to admins to default AdminSet
+      # Default admin set has a nil creating_user; guard against that condition
```

Comment:
The guard clause was on line 96. I don't think it makes sense to move this comment here.